src/Essentials/src/SemanticScreenReader/SemanticScreenReader.uwp.cs

```
@@ -12,6 +12,12 @@ public void Announce(string text)
 			if (WindowStateManager.Default.GetActiveWindow() is not Window window)
 				return;
 
+			if (window.Content is null)
+			{
+				// If the window content is null, we can't announce anything yet.
+				// This can happen if the app is still starting up.
+				return;
```

This doesn’t exactly fix the issue in question, right?

We’re just avoiding the crash when window.Content is null. But this ends up silently skipping the announcement if the visual tree isn’t ready. Now the user gets no feedback, the announcement just doesn’t happen. No crash, no warning, just silence.

From what I understand, on Windows, we need to locate a control with an automation peer to perform the announce operation. If that’s the expected behavior, maybe we shouldn’t silently skip here.

Instead, could we consider throwing a meaningful exception with a helpful message, so users know they’re calling Announce() too early and can fix it by moving the call to a better place?

---

@bhavanesh2001  Throwing an exception in an accessibility scenario is not an appropriate approach. Accessibility features are intended to enhance the user experience, not disrupt it. If a screen reader announcement fails, the application should continue functioning normally.

A better approach would be to return silently along with logging the issue using diagnostic tools such as Debug.WriteLine. This ensures that the failure is recorded for developers without affecting the user experience. In the Essentials project, diagnostic logging is commonly used to capture such errors without interrupting the app’s flow.
https://github.com/dotnet/maui/blob/main/src/Essentials/src/Permissions/Permissions.ios.cs#L38

@PureWeen, your thoughts?

---

@sheiksyedm  Just to clarify ,I wasn’t suggesting we should throw on end users’ devices. My point was to help devs catch the issue early, rather than silently skipping the announcement without any signal.

By users, I meant devs using the API, not app users.